src/search_datastructures.hpp

```diff
@@ -505,14 +509,20 @@ class LocalDataHolder
     // regarding the gathering of stats
     StatsHolder         stats{};
 
+    // search work space
+    std::vector<std::vector<uint8_t>> queries;
```

Comment:
Like I said, ideally we would use views instead. If we cannot, I would propose to use `TCDStringSet<std::vector<uint8_t>>` instead, because it avoid dynamic reallocations (just one buffer that is reused).

Reply:
I am assuming that LocalDataHolder is instantiated once per thread (I hope this is true, otherwise I broke the code).
The search function currently takes a std::vector<std::vector<uint8_t>>. It is also only allocated once when creating LocalDataHolder. This could be removed if I switch over to ranges in my library.

Reply:
Yeah, I would really prefer if we could get rid of this. Is it feasible time-wise to change this in your library? Like I said, you could do some very simple checks for begin() and end() without going all crazy with concepts...

If it is not feasible, I would still prefer if we can rename this to `queries_as_ranks` or something like that. And also attach a `TODO remove this as soon as fmindex_collection can handle ranges`